Upstream-Status: Backport

This is required for us to properly change arm_arch via the command line to
the value set in BitBake.

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From 99cae1ee0ee6c49c0c3d276dbbb8a271d9a03bb4 Mon Sep 17 00:00:00 2001
From: hashimoto <hashimoto@chromium.org>
Date: Thu, 1 Dec 2016 22:14:34 -0800
Subject: [PATCH] gn: Make arm_arch configurable

Chrome OS needs to use values like armv7ve and armv8-a+crc.

BUG=670242

Review-Url: https://codereview.chromium.org/2543953002
Cr-Commit-Position: refs/heads/master@{#435867}
---
 build/config/arm.gni | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/build/config/arm.gni b/build/config/arm.gni
index 27ed7738abce..47c836a22219 100644
--- a/build/config/arm.gni
+++ b/build/config/arm.gni
@@ -14,6 +14,10 @@ if (current_cpu == "arm" || v8_current_cpu == "arm") {
     # platforms.
     arm_version = 7
 
+    # The ARM architecture. This will be a string like "armv6" or "armv7-a".
+    # An empty string means to use the default for the arm_version.
+    arm_arch = ""
+
     # The ARM floating point mode. This is either the string "hard", "soft", or
     # "softfp". An empty string means to use the default one for the
     # arm_version.
@@ -48,7 +52,9 @@ if (current_cpu == "arm" || v8_current_cpu == "arm") {
   }
 
   if (arm_version == 6) {
-    arm_arch = "armv6"
+    if (arm_arch == "") {
+      arm_arch = "armv6"
+    }
     if (arm_tune != "") {
       arm_tune = ""
     }
@@ -58,7 +64,9 @@ if (current_cpu == "arm" || v8_current_cpu == "arm") {
     arm_fpu = "vfp"
     arm_use_thumb = false
   } else if (arm_version == 7) {
-    arm_arch = "armv7-a"
+    if (arm_arch == "") {
+      arm_arch = "armv7-a"
+    }
     if (arm_tune == "") {
       arm_tune = "generic-armv7-a"
     }
@@ -80,7 +88,9 @@ if (current_cpu == "arm" || v8_current_cpu == "arm") {
       arm_fpu = "vfpv3-d16"
     }
   } else if (arm_version == 8) {
-    arm_arch = "armv8-a"
+    if (arm_arch == "") {
+      arm_arch = "armv8-a"
+    }
     if (arm_tune == "") {
       arm_tune = "generic-armv8-a"
     }
-- 
2.13.6

